src/components/StudentProfile.js

```diff
@@ -0,0 +1,336 @@
+            <Form.Group controlId="formBasicDate">
+              <Form.Label>Select a date: </Form.Label>
+              <br/>
+              <DatePicker
+                    selected={date}
+                    onChange={handleChange}
+                    minDate={today}
+                    maxDate={in30Days}
+                    showTimeSelect
+                  />
+            </Form.Group>
+            
+            
+
+            <Form.Group controlId="formBasicStyle">
+              <FormLabel>Pick a style!</FormLabel>
+              <Dropdown
+                value={style}
+                onSelect={(eventKey) => setStyle(eventKey)}
+              >
+                <Dropdown.Toggle variant="success" id="dropdown-basic">
+                  Style
+                </Dropdown.Toggle>
+
+                <Dropdown.Menu>
+                  <Dropdown.Item eventKey="Hatha">Hatha</Dropdown.Item>
+                  <Dropdown.Item eventKey="Vinyasa">Vinyasa</Dropdown.Item>
+                  <Dropdown.Item eventKey="Yin">Yin Yoga</Dropdown.Item>
+                  <Dropdown.Item eventKey="Restorative">
+                    Restorative
+                  </Dropdown.Item>
+                  <Dropdown.Item eventKey="Ashtanga">Ashtanga</Dropdown.Item>
+                  <Dropdown.Item eventKey="Power">Power Yoga</Dropdown.Item>
+                  <Dropdown.Item eventKey="Pre-Natal">Pre-natal</Dropdown.Item>
+                </Dropdown.Menu>
+              </Dropdown>
+            </Form.Group>
+
+            <Button variant="dark" onClick={() => handleSearch()}>
+              Search
+            </Button>
+          </Form>
+          <Card style={{marginBottom: 20}}>
+            {yogaClass.map((yogaClass, index) => (
+              <React.Fragment>
+                <Card.Header>{yogaClass.title} </Card.Header>
+
+                <Card.Body>
+                  <Card.Title>Level:{yogaClass.level} </Card.Title>
+                  <ListGroup variant="flush">
+                    <ListGroup.Item>
+                      Teacher: {yogaClass.first_name} {yogaClass.last_name}
+                    </ListGroup.Item>
+                    <ListGroup.Item>
+                      Years of experience: {yogaClass.years_experience}
+                    </ListGroup.Item>
+                    <ListGroup.Item> Time: {yogaClass.time}</ListGroup.Item>
+                    <ListGroup.Item>Price: ${yogaClass.price}</ListGroup.Item>
+                    <ListGroup.Item>Date: {yogaClass.date}</ListGroup.Item>
+                    <ListGroup.Item>Style: {yogaClass.style}</ListGroup.Item>
+                    <ListGroup.Item>
+                      Duration: {yogaClass.duration} minutes
+                    </ListGroup.Item>
+                  </ListGroup>
+                  <Card.Text> Description: {yogaClass.description}</Card.Text>
+                  <Button variant="dark" onClick={() => handleSearch()}></Button>  {/* can u use the same book for async? */}
```

Comment:
What does this button do? you are calling search again?

Reply:
Calling book